tests/savepoint/test_translate.py

```diff
@@ -94,25 +106,46 @@ def sample_wherefail(
     computed_failures = computed_data[found_indices]
     reference_failures = ref_data[found_indices]
 
+    # List all errors
     return_strings = []
     bad_indices_count = len(found_indices[0])
     if print_failures:
         for b in range(0, bad_indices_count, failure_stride):
             full_index = [f[b] for f in found_indices]
+            abs_err = abs(computed_failures[b] - reference_failures[b])
+            metric_err = compare_scalar(computed_failures[b], reference_failures[b])
             return_strings.append(
                 f"index: {full_index}, computed {computed_failures[b]}, "
                 f"reference {reference_failures[b]}, "
-                f"diff {abs(computed_failures[b] - reference_failures[b])}"
+                f"absolute diff {abs_err:.3e}, "
+                f"metric diff: {metric_err:.3e}"
             )
-    sample = [f[0] for f in found_indices]
-    fullcount = len(ref_data.flatten())
 
+    # Determine worst result
+    worst_err = 0
+    worst_idx = 0
+    worst_full_idx = 0
+    for b in range(0, bad_indices_count, failure_stride):
+        err = compare_scalar(computed_failures[b], reference_failures[b])
+        if worst_err < err:
```

Comment:
(optional) you can also use `numpy.argmax` and `numpy.unravel_index` to find the largest error index instead of having a loop over the failures.
```
err =  compare_arr(computed_data, ref_data)
worst_err = err.max()
worst_full_idx = np. unravel_index(np.argmax(err, axis=None), err.shape)
```
I think you won't need the scalar version?